src/woo_publications/publications/tests/test_process_source_document_task.py

```diff
@@ -0,0 +1,263 @@
+                json={
+                    "inhoud": base64.b64encode(b"12345").decode("ascii"),
+                    "bestandsnaam": "updated.txt",
+                    "bestandsomvang": 5,
+                    "lock": lock,
+                },
+            )
+            client.unlock_document(uuid=oz_uuid, lock=lock)
+
+        source_url = (
+            "http://openzaak.docker.internal:8001/documenten/api/v1/"
+            f"enkelvoudiginformatieobjecten/{oz_uuid}?versie=1"
+        )
+        woo_document = DocumentFactory.create(source_url=source_url)
+
+        process_source_document(
+            document_id=woo_document.id, base_url="http://host.docker.internal:8000/"
+        )
+
+        woo_document.refresh_from_db()
+        self.assertTrue(woo_document.upload_complete)
+        # values from version 1
+        self.assertEqual(woo_document.creatiedatum, date(2025, 1, 1))
+        self.assertEqual(woo_document.bestandsformaat, "text/plain")
+        self.assertEqual(woo_document.bestandsnaam, "data.txt")
+        self.assertEqual(woo_document.bestandsomvang, 10)
+
+        with (
+            self.subTest("verify content from version 1 is copied"),
+            get_client(self.service) as client,
+        ):
+            download = client.get(
+                f"enkelvoudiginformatieobjecten/{woo_document.document_uuid}/download"
+            )
+
+            assert download.status_code == 200
+            self.assertEqual(download.content, b"aaaaaaaaaa")
+
+    def test_document_creation_is_idempotent(self):
+        with get_client(self.service) as client:
+            oz_uuid = _create_initial_document(
+                client=client,
+                creation_date=date(2025, 1, 1),
+                content=b"a" * 10,
+                filename="data.txt",
+                content_type="text/plain",
+            )
+
+        source_url = (
+            "http://openzaak.docker.internal:8001/documenten/api/v1/"
+            f"enkelvoudiginformatieobjecten/{oz_uuid}"
+        )
+        woo_document = DocumentFactory.create(source_url=source_url)
+
+        registered_document_uuids: set[str] = set()
+        for run in range(1):
+            with self.subTest(processing_run=run):
+                process_source_document(
+                    document_id=woo_document.id,
+                    base_url="http://host.docker.internal:8000/",
+                )
+
+                woo_document.refresh_from_db()
+                self.assertTrue(woo_document.upload_complete)
+                registered_document_uuids.add(str(woo_document.document_uuid))
+                # we expect the document to be registered only once
+                self.assertEqual(len(registered_document_uuids), 1)
+
+    def test_file_parts_failure_is_recoverable(self):
+        # Simulate that something went wrong while uploading the file parts, and
+        # retrying the task can pick up where it left of.
```

Comment:
The test name seems to be misleading to me. Because we aren't testing that during the part upload something midway the process went wrong and re-running this code will resume from where we left off.

This code tests only that the code runs fine when the file parts are set beforehand. Not when part 1 has uploaded correctly but 2 and 3 didn't.